web/src/message_view_header.ts

```diff
@@ -115,10 +115,17 @@ function get_message_view_header_context(filter: Filter | undefined): MessageVie
         }
 
         if (inbox_util.is_visible() && inbox_util.is_channel_view()) {
-            const stream_name_with_privacy_symbol_html = render_inline_decorated_channel_name({
+            let stream_name_with_privacy_symbol_html = render_inline_decorated_channel_name({
                 stream: current_stream,
                 show_colored_icon: true,
             });
+            // At different font sizes, the icon and channel name
+            // can have different spacing. This class fixes that.
+            const index = stream_name_with_privacy_symbol_html.indexOf("zulip-icon");
+            stream_name_with_privacy_symbol_html =
+                stream_name_with_privacy_symbol_html.slice(0, index) +
+                "fix-channel-name-and-icon-spacing " +
+                stream_name_with_privacy_symbol_html.slice(index);
```

Comment:
Should we introduce a variable in `render_inline_decorated_channel_name` instead? I thought there were already too many variables there.